compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -597,6 +616,19 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         id
     }
 
+    /// This checks if an item is a `#[macro_use]` macro, and thus always
+    /// defined at the top level. As a special case, if the macro is already
+    /// at the top level, it remains at its current level.
+    fn check_force_top_level(&self, item: &Item) -> bool {
+        if self.is_at_top_level {
```

Comment:
I'm pretty sure it isn't. If you go look at the cases where I set `is_at_top_level`, I don't think all of them are covered by checking `self.current_hir_id_owner.0`. I think I tried that at some point and it didn't work. It's somewhat irrelevant though, because if we don't move the exported macros to the crate root all of this code disappears.